src/System.Data.SqlClient/tests/FunctionalTests/TcpDefaultForAzureTest.cs

```diff
@@ -36,7 +36,14 @@ static TcpDefaultForAzureTest()
         public static void NonAzureNoProtocolConnectionTest()
         {
             builder.DataSource = InvalidHostname;
-            CheckConnectionFailure(builder.ConnectionString, null); // Managed and Native SNI have different default protocols for plain server names
+            if(ManualTesting.Tests.DataTestUtility.IsUsingManagedSNI())
+            {
+                CheckConnectionFailure(builder.ConnectionString, TCP);
+            }
+            else
+            {
+                CheckConnectionFailure(builder.ConnectionString, NP);
+            }
```

Comment:
Nit: could also be:
```C#
CheckConnectionFailure(builder.ConnectionString,
    ManualTesting.Tests.DataTestUtility.IsUsingManagedSNI() ? TCP : NP);
```
to better highlight what's actually different between the two.